ligo/stablecoin/actions.ligo

```diff
@@ -240,4 +333,232 @@ function is_operator_action
+ * minter is not present. Fails if sender is not master minter.
+ *)
+function remove_minter
+  ( const parameter : remove_minter_params
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_master_minter (store)
+; case store.minting_allowances[parameter] of
+    Some (u) -> skip
+  | None -> failwith ("Requested address is not minter")
+  end
+} with
+  ( (nil : list (operation))
+  , store with record
+      [ minting_allowances = Big_map.remove
+          ( parameter
+          , store.minting_allowances
+          )
+      ]
+  )
+
+(*
+ * Produces tokens to a wallet associated with the given address
+ *)
+function mint
+  ( const parameters : mint_params
+  ; const store      : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+
+; const senderAddress : address = Tezos.sender
+
+; function mint_tokens
+    ( const accumulator       : storage
+    ; const mint_param        : mint_param
+    ; const current_allowance : nat
+    ) : storage is block
+  { const unwrapped_parameter : mint_param_ =
+      Layout.convert_from_right_comb ((mint_param : mint_param))
+  ; fail_on
+      ( unwrapped_parameter.amount > current_allowance
+      , "Requested amount of minting tokens exceeds current allowance"
+      )
+
+  ; const updated_allowance : nat =
+      abs (current_allowance - unwrapped_parameter.amount)
+  ; const updated_allowances : minting_allowances =
+      Big_map.update (Tezos.sender, Some (updated_allowance), accumulator.minting_allowances)
+  ; const updated_store : storage = accumulator with record
+      [ minting_allowances = updated_allowances
+      ; total_supply = accumulator.total_supply + unwrapped_parameter.amount
+      ]
+  } with credit_to (unwrapped_parameter, updated_store)
+
+} with
+  ( (nil : list (operation))
+  , List.fold
+      ( function
+          ( const accumulator : storage
+          ; const mint_param  : mint_param
+          ) : storage is case accumulator.minting_allowances[senderAddress] of
+            None -> (failwith ("Sender address is not minter") : storage)
+          | Some (current_allowance) ->
+              mint_tokens (accumulator, mint_param, current_allowance)
+          end
+      , parameters
+      , store
+      )
+  )
+
+(*
+ * Decreases balance and total supply of tokens by the given amount
+ *)
+function burn
+  ( const parameters : burn_params
+  ; const store      : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+; const sender_address : address = Tezos.sender
+; function burn_tokens
+    ( const accumulator : storage
+    ; const burn_amount : nat
+    ) : storage is block
+  { const debited_storage : storage = debit_from
+      ( record
+          [ from_ = sender_address
+          ; amount = burn_amount
+          ]
+      , accumulator
+      )
+  } with debited_storage with record
+      [ total_supply = abs (debited_storage.total_supply - burn_amount) ]
```

Comment:
Why is `abs` required here?

Reply:
This error won't be called anyway since in that case `debit_from` would fail earlier, but anyway I've changed it

Reply:
Yes, of course. We woulndn't encounter that case normally, unless there is some bug else where. But if there is a bug else where and we indeed hit this case, having `abs` function here will result in silently corrupted contract state, right, that is of course unless our tests don't catch it. 

But I think it is better not to even leave that possibility in code..